### DISCUSSION TITLE
Generate different image sizes (formats) for existing media after migrating to new media library

### DISCUSSION THREAD

**Discussion by tomolenet** [strapi/strapi#7794] [Legacy Topics]
Hi, I recently migrated to the new media library. But in the [migration guide](https://strapi.io/documentation/3.0.0-beta.x/migration-guide/migration-guide-beta.19-to-beta.20.html) I am missing a script or something to migrate the existing media to make use of the formats attribute. The existing media has no attribute `formats`. If I upload new media the `formats` attribute is available for this media.

```
{
   ...
    "formats": {
      "thumbnail": {
        "name": "thumbnail_imm031_32.jpg",
        "ext": ".jpg",
        ...
        "url": "/uploads/thumbnail_imm031_32_fcf93ff3c0.jpg"
      },
      "large": {
        "name": "large_imm031_32.jpg",
        "ext": ".jpg",
        ...
        "url": "/uploads/large_imm031_32_fcf93ff3c0.jpg"
      },
      "medium": {
        "name": "medium_imm031_32.jpg",
        "ext": ".jpg",
        ...
        "url": "/uploads/medium_imm031_32_fcf93ff3c0.jpg"
      },
      "small": {
        "name": "small_imm031_32.jpg",
        "ext": ".jpg",
        ...
        "url": "/uploads/small_imm031_32_fcf93ff3c0.jpg"
      }
    },
   ...
}
```

**Comment by derrickmehaffy** [comment]
Currently we do not have a script or method to generate the `formats` for older media, the media will have to be reuploaded/replaced and during this time the formats will be generated.
